packages/commerce-sdk-react/src/hooks/Test/query.ts

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
+import {ApiClients, Argument, DataType} from '../types'
+import useCommerceApi from '../useCommerceApi'
+import {useQuery} from '../useQuery'
+
+type Client = ApiClients['shopperBaskets']
+
+/**
+ * A hook for `ShopperBaskets#getBasket`.
+ * Gets a basket.
+ * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=getBasket} for more information about the API endpoint.
+ * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperbaskets.shopperbaskets-1.html#getbasket} for more information on the parameters and returned data type.
+ * @returns An object describing the state of the request.
+ */
+export const useBasket = (
+    apiOptions: Argument<Client['getBasket']>,
+    queryOptions: UseQueryOptions<DataType<Client['getBasket']>>
+): UseQueryResult<DataType<Client['getBasket']>> => {
+    const {shopperBaskets: client} = useCommerceApi()
+    const method = (arg: Argument<Client['getBasket']>) => client.getBasket(arg)
+    const requiredParameters = ['organizationId', 'basketId', 'siteId'] as const
+    // Parameters can be set in `apiOptions` or `client.clientConfig`; they are merged in the helper
+    // hook, so we use a callback here that receives that merged object.
+    const getQueryKey = <T extends Record<string, unknown>>(parameters: T) =>
+        [
+            'https://',
+            parameters.shortCode,
+            '.api.commercecloud.salesforce.com/checkout/shopper-baskets/',
+            parameters.version,
+            '/organizations/',
+            parameters.organizationId,
+            '/baskets/',
+            parameters.basketId,
+            '?',
+            'siteId',
+            parameters.siteId,
+            'locale',
+            parameters.locale,
+            // Full parameters last for easy lookup
+            parameters
+        ] as const
```

Comment:
I used the full generated URL here as the query key. I can't imagine `shortCode` or `version` ever changing, so we probably don't need the base URI part. Definitely need path and query parameters. Not sure what the optimal format is, though.